src/python/bcc/table.py

```diff
@@ -559,6 +561,58 @@ def __delitem__(self, key):
             lib.bpf_close_perf_event_fd(self._open_key_fds[key])
         del self._open_key_fds[key]
 
+    def _get_event_class(self):
+        ct_mapping = { 'char': ct.c_char,
+                       'unsigned char': ct.c_ubyte,
+                       'u8': ct.c_ubyte,
+                       'u8 *': ct.c_char_p,
+                       'char *': ct.c_char_p,
+                       'short': ct.c_short,
+                       'unsigned short': ct.c_ushort,
+                       'u16': ct.c_ushort,
+                       'int': ct.c_int,
+                       'unsigned int': ct.c_uint,
+                       'u32': ct.c_uint,
+                       'long': ct.c_long,
+                       'unsigned long': ct.c_ulong,
+                       'long long': ct.c_longlong,
+                       'unsigned long long': ct.c_ulonglong,
+                       'u64': ct.c_ulonglong,
+                       'void *': ct.c_void_p }
+        fields = []
+        prog = re.compile(r"(.+) \[([0-9]+)\]$")
+
+        num_fields = lib.bpf_perf_event_fields(self.bpf.module)
+        i = 0
+        while i < num_fields:
+            field_name = lib.bpf_perf_event_field_name(self.bpf.module, i)
+            field_type = lib.bpf_perf_event_field_type(self.bpf.module, i)
```

Comment:
Is it possible to have just one libbcc call like `lib.bpf_perf_event_field` to get field name and type in one shot?

Reply:
Ok. I can concatenate type and name in libbcc and then split them in Python.